### PR TITLE
all: remove use io/ioutil deprecated package & fix some deprecated thing

### PR DESCRIPTION
**Pull request by zchee**
SSIA.

- `ioutil.ReadDir` -> `os..ReadDir`
- `ioutil.ReadFile` -> `os.ReadFile`
- `ioutil.TempFile` -> `os.CreateTemp`
- `ioutil.TempDir` -> `os.MkdirTemp`
- `ioutil.Discard` -> `io.Discard`
- `ioutil.ReadAll` -> `io.ReadAll`
- `os.SEEK_CUR` -> `io.SeekCurrent`

### CODE REVIEW

**Review by ayushr2 — APPROVED** [review]

File: pkg/seccomp/seccomp_test.go [hunk 1]
```diff
@@ -1078,7 +1076,6 @@ func TestBasic(t *testing.T) {
 
 // TestRandom tests that randomly generated rules are encoded correctly.
 func TestRandom(t *testing.T) {
-	rand.Seed(time.Now().UnixNano())
```
Comment (ayushr2) [inline pkg/seccomp/seccomp_test.go:1081]: Why is this removed?

Comment (zchee) [reply pkg/seccomp/seccomp_test.go:1081]: Just https://pkg.go.dev/math/rand#Seed
